File: android/src/main/kotlin/me/yohom/amapbase/search/SearchFunctionRegistry.kt
package me.yohom.amapbase.search

import me.yohom.amapbase.search.handler.fetchdata.*
import me.yohom.amapbase.search.handler.routeplan.CalculateDriveRoute

val SEARCH_METHOD_HANDLER: Map<String, SearchMethodHandler> = mapOf(
        "search#calculateDriveRoute" to CalculateDriveRoute,
        "search#searchPoi" to SearchPoiKeyword,
        "search#searchPoiBound" to SearchPoiBound,
        "search#searchPoiPolygon" to SearchPoiPolygon,
        "search#searchPoiId" to SearchPoiId,
        "search#searchRoutePoiLine" to SearchRoutePoiLine,
        "search#searchRoutePoiPolygon" to SearchRoutePoiPolygon
)

File: ios/Classes/search/SearchFunctionRegistry.m
//
// Created by Yohom Bao on 2018-12-11.
//

#import "SearchFunctionRegistry.h"
#import "SearchMethodHandler.h"
#import "SearchRoutePoiPolygon.h"
#import "CalculateDriveRoute.h"
#import "SearchPoiKeyword.h"
#import "SearchPoiBound.h"
#import "SearchPoiPolygon.h"
#import "SearchPoiId.h"
#import "SearchRoutePoiLine.h"


static NSDictionary<NSString *, NSObject <SearchMethodHandler> *> *_map;

@implementation SearchFunctionRegistry {

}
+ (NSDictionary<NSString *, NSObject <SearchMethodHandler> *> *)searchMethodHandler {
    if (!_map) {
        _map = @{
                @"search#calculateDriveRoute": [CalculateDriveRoute alloc],
                @"search#searchPoi": [SearchPoiKeyword alloc],
                @"search#searchPoiBound": [SearchPoiBound alloc],
                @"search#searchPoiPolygon": [SearchPoiPolygon alloc],
                @"search#searchPoiId": [SearchPoiId alloc],
                @"search#searchRoutePoiLine": [SearchRoutePoiLine alloc],
                @"search#searchRoutePoiPolygon": [SearchRoutePoiPolygon alloc],
        };
    }
    return _map;
}

@end